src/login/logind-dbus.c

```diff
@@ -893,6 +905,8 @@ static int method_create_session(sd_bus_message *message, void *userdata, sd_bus
         if (r < 0)
                 goto fail;
 
+        session->leader_pidfd = dup(pidfd);
```

Comment:
dup() is not ok in our codebase. use `fcntl(fd, FD_DUPFD_CLOEXEC, 3)` instead.

also, check if pidfd >= 0 first, and set to -EBADF otherwise

Reply:
I actually got rid of this after migration to PidRef.